models/datastore.go

```diff
@@ -24,18 +25,48 @@ func (r ResourceType) String() string {
 }
 
 type ResourceSpec struct {
-	ID        uuid.UUID
-	Version   int
-	Name      string
-	Type      ResourceType
+	// ID represents the unique identifier of a resource.
```

Comment:
do we need comments here we can have the comments on the equals method that should be sufficient

Reply:
updated as mentioned